isis/scripts/findfeaturesSegment

```diff
@@ -0,0 +1,237 @@
+    if (params.get("debug") == "True"):
+        log.basicConfig(level=logging.DEBUG)
+    else: 
+        log.basicConfig(level=logging.INFO)    
+
+    if "nl" not in params : 
+        raise ValueError('**User Error** invalid command Line. Missing Parameter "nl"')
+
+    img_list = []
+    if "from" in params:
+        img_list = [params["from"]]
+    elif "fromlist" in params:
+        img_list = read_cubelist(params["fromlist"])
+    else:
+        raise ValueError('**User Error** invalid command Line. Missing Parameter "nl"')    
+
+    from_segments = {}
+
+    for img in img_list:
+        res = segment(img)
+        from_segments[img] = res
+        
+    match_segments = segment(params["match"])
+    max_from_segments = max([len(v) for k,v in from_segments.items()])
+
+    onets = []
+    for i in range(len(match_segments)):
+        for j in range(max_from_segments): 
+            match_segment_n = i+1
+            from_segment_n = j+1
+            
+            new_params = deepcopy(params)
+
+            # make sure none of these keys are still in the params
+            new_params.pop("fromlist", None)
+            new_params.pop("from", None)
+
+            new_params["match"] = match_segments[i]["Path"]
+
+            og_onet = Path(params["onet"])
+            og_tolist = Path(params["tolist"])
+            og_pointid = params["pointid"]
+            og_networkid = params["networkid"]
+            
+            from_images = [segment[j]["Path"] for image, segment in from_segments.items() if j < len(segment)]
+            starting_lines = [segment[j]["StartingLine"] for image, segment in from_segments.items() if j < len(segment)]
+            
+            fromlist_path = Path(og_tolist.parent / f"from_images_segment{from_segment_n}.lis")
+            log.debug(f"fromlist: {from_images}")
+            
+            if not fromlist_path.exists():
+                log.debug(f"writing to: {fromlist_path}")
+                kisis.fromlist.make(from_images, fromlist_path)
+            else:
+                log.debug(f"{fromlist_path} already exists")
+            
+            if "debuglog" in new_params:
+                og_log =  Path(new_params["debuglog"])
+                new_log = og_log.parent / f"{og_log.stem}.match_segment{match_segment_n}.from_segment{from_segment_n}{og_log.suffix}" 
+                new_params["debuglog"] = str(new_log)
+                
+            new_params["fromlist"] = str(fromlist_path)
+            new_params["networkid"] = og_networkid + f"_match_segment{match_segment_n}_from_segment{from_segment_n}"
+            new_params["pointid"] = og_pointid + f"_match_segment{match_segment_n}_from_segment{from_segment_n}"
+            new_params["onet"] = f"{og_onet.parent/og_onet.stem}.match_segment{match_segment_n}.from_segment{from_segment_n}.net"
+            new_params["tolist"] = f"{og_tolist.parent/og_tolist.stem}.match_segment{match_segment_n}.from_segment{from_segment_n}.lis"
```

Comment:
These should probably be changed to include the image names, not just match/from and segment number